From c328aef74ca5ddd70005ee898e219690720c0cb3 Mon Sep 17 00:00:00 2001
From: Lei Xu <lei@cloudera.com>
Date: Thu, 14 Jan 2016 11:36:30 -0800
Subject: [PATCH 1018/1023] CLOUDERA-BUILD. Fix
 TestOfflineImageViewerForAcl.testPBImageXmlWriterForAcl
 Premature EOF. (CDH-36217)

Change-Id: I89098db68e13d19744919559df883bea39120b3d
(cherry picked from commit 0fc1c9456ab6d7a933c19b2ff141cbabed7902cf)
---
 .../TestOfflineImageViewerForAcl.java              |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/TestOfflineImageViewerForAcl.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/TestOfflineImageViewerForAcl.java
index bb7e191..1e46e00 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/TestOfflineImageViewerForAcl.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/TestOfflineImageViewerForAcl.java
@@ -222,6 +222,7 @@ public void testPBImageXmlWriterForAcl() throws Exception{
     PrintWriter o = new PrintWriter(output);
     PBImageXmlWriter v = new PBImageXmlWriter(new Configuration(), o);
     v.visit(new RandomAccessFile(originalFsimage, "r"));
+    o.flush();
     SAXParserFactory spf = SAXParserFactory.newInstance();
     SAXParser parser = spf.newSAXParser();
     final String xml = output.toString();
-- 
1.7.9.5

